src/Bind/BindDefinition.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace Nuwave\Lighthouse\Bind;
+
+use GraphQL\Language\AST\FieldDefinitionNode;
+use GraphQL\Language\AST\InputObjectTypeDefinitionNode;
+use GraphQL\Language\AST\InputValueDefinitionNode;
+use GraphQL\Type\Definition\Type;
+use Illuminate\Database\Eloquent\Model;
+use Nuwave\Lighthouse\Exceptions\DefinitionException;
+use Nuwave\Lighthouse\Schema\AST\ASTHelper;
+
+/**
+ * @template-covariant TClass of object
+ * @property-read class-string<TClass> $class
+ * @property-read string $column
+ * @property-read array<string> $with
+ * @property-read bool $optional
```

Comment:
I don't see why those annotations would be necessary - the last one seems wrong even.